src/main/java/app/dbService/PostgresBookService.java

```diff
@@ -102,15 +109,18 @@ public void updateBook(Book book) throws DBException {
             Connection connection = getPostgresConnection();
             if (connection != null) {
                 connection.setAutoCommit(false);
+            } else {
+                throw new NullPointerException();
```

Comment:
throwing your custom Exception/moving everything that uses connection variable into if would be a lot better in terms of maintainability and code quality